nasa/harmony
# HARMONY-1426: Make changes to improve performance to scale up more pods

## Jira Issue ID
HARMONY-1426

## Description
After performing testing with a workload run and with a single large 110K granule request we identified some changes to improve throughput and also help ensure synchronous requests can complete in reasonable times while large requests are also running.

## Local Test Steps
You will need to change your .env file for running harmony to change the names of the queues for work item updates since they are now FIFO:
```
WORK_ITEM_UPDATE_QUEUE_URL=http://localhost:4566/queue/work-item-update-queue.fifo
LARGE_WORK_ITEM_UPDATE_QUEUE_URL=http://localhost:4566/queue/large-work-item-update-queue.fifo
```
1. Delete your localstack pod: `kubectl delete pod -n harmony localstack-pod`
2. `bin/start-postgres-localstack` (this will cause the FIFO queues to be created for work item updates)
3. Build the scheduler (I think I changed some scheduler code) `npm run build` in the `kubernetes-services/work-scheduler/` directory.
4. `bin/restart-services`

Make sure requests complete successfully for you.

In sandbox I was able to run with 200 pods and the 110K granule request completed in under four hours. Queues never backed up to more than 60 items for the work item updates and while there were some slower DB queries in locking the jobs table with these changes and the 200 pod concurrency the locking was in the 500-700ms range as opposed to 2 minutes prior to the changes.

## PR Acceptance Checklist
